spec/support/request_helpers/event_logging.rb

```diff
@@ -8,8 +8,7 @@ def logs_event(event_class_name, expected_payload:)
         expect(Event.first.action).to eq(event_class_name)
         expect(Event.first.user_uid).to eq(nil)
 
-        expected_payload = expected_payload.deep_stringify_keys.merge("base_path" => base_path)
-        expect(Event.first.payload).to eq(expected_payload)
+        expect(Event.first.payload).to eq(instance_exec(&expected_payload_proc))
```

Comment:
Why does this need to use `instance_exec` here as opposed to just calling the proc?